springfox-core/src/main/java/springfox/documentation/service/ResolvedMethodParameter.java

```diff
@@ -31,20 +33,31 @@
 
 public class ResolvedMethodParameter {
   private final int parameterIndex;
+  private final boolean returnType;
   private final List<Annotation> annotations;
   private final String defaultName;
   private final ResolvedType parameterType;
 
   public ResolvedMethodParameter(String paramName, MethodParameter methodParameter, ResolvedType parameterType) {
     this(methodParameter.getParameterIndex(),
+        false,
         paramName,
         newArrayList(methodParameter.getParameterAnnotations()),
         parameterType);
   }
+  
+  public ResolvedMethodParameter(ResolvedType parameterType) {
+    this(0,
+        true,
+        "",
+        newArrayList(new Annotation[0]),
+        parameterType);
+  }
 
-  public ResolvedMethodParameter(int parameterIndex, String defaultName, List<Annotation> annotations, ResolvedType
+  private ResolvedMethodParameter(int parameterIndex, boolean returnType, String defaultName, List<Annotation> annotations, ResolvedType
```

Comment:
Sure lets chat about that to do it the right way. This change is subverting the intent of this class tho'